### ISSUE TITLE
Not useful error MT2301

### ISSUE THREAD

**Issue by marek-safar** [xamarin/xamarin-macios#12635]
Building with the recent RC2 preview I get the following error message which is not very helpful

```
ILLINK : error MT2301: The linker step 'Setup' failed during processing: This version of Microsoft.iOS requires the iOS 15.0 SDK (shipped with Xcode 13.0). Either upgrade Xcode to get the required header files or set the managed linker behaviour to Link Framework SDKs Only in your project's iOS Build Options > Linker Behavior (to try to avoid the new APIs). [/Users/xyz/git/temp/xam-net6/xam-net6.csproj]
```

As far as I can tell Microsoft.iOS runs linker in "copy" mode for simulators all the time. I'm also building using VS Code which does not have anything like "Options > Linker Behaviour". There is also no trace of real exception even if I enabled diagnostics logging to see what is happening.

**Comment by rolfbjarne** [comment]
> Building with the recent RC2 preview I get the following error message which is not very helpful

Agreed it's not very helpful, so I'm keeping this issue open to figure out how to make it better.

The root cause is that you don't have the right version of Xcode 13 installed.

> I'm also building using VS Code which does not have anything like "Options > Linker Behaviour".

I'm not sure how we can improve this. We try to give a helpful message for the majority of our users, and in the past our majority has used VS(Mac). In the IDE it was also possible to double-click the error, and the IDE would open a web page with a more detailed explanation. Neither of these options apply to VSCode, but how can we detect that? Do we even want to try?

We could just remove the helpful text, but that seems like a lose-lose scenario.

> There is also no trace of real exception even if I enabled diagnostics logging to see what is happening.

Exactly how do you enable diagnostics logging? Passing /v:diag to dotnet? If so, afaik there's no way for our custom linker step to know that diagnostic logging was enabled. We can emit warnings and errors, and that's it.

**Comment by marek-safar** [comment]
> The root cause is that you don't have the right version of Xcode 13 installed.

Correct but then why confuse the user with suggestions to change linker behaviour? If the problem is in custom steps then the Linker Behaviour is not relevant.

> Exactly how do you enable diagnostics logging? Passing /v:diag to dotnet

Yes. I think you could use `MessageContainer.CreateDiagnosticMessage` for that or we could move to logic to core linker and log crashes from custom steps there.

**Comment by rolfbjarne** [comment]
Fixed in #22800.
